### PR TITLE
bugfix: last page not getting added to results

### PR DESCRIPTION
**Pull request by k-funk**
Resolves https://github.com/munir131/attachment-downloader/issues/32

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -257,15 +257,15 @@ function getListOfMailIdByLabel(auth, labelId, maxResults = 500, nextPageToken)
...
+      if (response.data) {
         messageIds = messageIds.concat(response.data.messages)
-        spinner.text = "Reading page: " + ++pageCounter
-        resolve(getListOfMailIdByLabel(auth, labelId, 500, response.data.nextPageToken))
-      } else {
-        messageIds = messageIds.concat(response.data.messages)
-        resolve(messageIds)
+        if (response.data.nextPageToken) {
+          spinner.text = "Reading page: " + ++pageCounter
+          resolve(getListOfMailIdByLabel(auth, labelId, maxResults, response.data.nextPageToken))
```
Comment (k-funk) [inline index.js:264]: maxResults shouldn't have been hardcoded to 500

File: index.js [hunk 2]
```diff
@@ -257,15 +257,15 @@ function getListOfMailIdByLabel(auth, labelId, maxResults = 500, nextPageToken)
         console.log('The API returned an error: ' + err);
         reject(err);
       }
-      if (response.data && response.data.nextPageToken) {
+      if (response.data) {
         messageIds = messageIds.concat(response.data.messages)
```
Comment (k-funk) [inline index.js:261]: messageIds should be concatenated even if there's no `nextPageToken`

File: index.js [hunk 3]
```diff
@@ -257,15 +257,15 @@ function getListOfMailIdByLabel(auth, labelId, maxResults = 500, nextPageToken)
...
-      } else {
-        messageIds = messageIds.concat(response.data.messages)
-        resolve(messageIds)
+        if (response.data.nextPageToken) {
+          spinner.text = "Reading page: " + ++pageCounter
+          resolve(getListOfMailIdByLabel(auth, labelId, maxResults, response.data.nextPageToken))
+        }
       }
-
+      spinner.text = "All pages are read"
```
Comment (k-funk) [inline index.js:267]: copy-pasted from `function getAllMails()`

**Comment by munir131** [comment]
@k-funk Thanks for PR

**Pull request merged**